source/developers/in-context-editing-ftl.rst

```diff
@@ -0,0 +1,368 @@
+:is-up-to-date: True
+
+:orphan:
+
+.. document does not appear in any toctree, this file is referenced
+   use :orphan: File-wide metadata option to get rid of WARNING: document isn't included in any toctree for now
+
+.. index:: Freemarker Templates In-Context Editing
+
+.. _in-context-editing-ftl:
+
+======================================
+Freemarker Template In-Context Editing
+======================================
+
+.. Highlighting language used is "guess" (let Pygments guess the lexer based on contents, only works with certain well-recognizable languages) since there's no Pygment lexer for freemarker
+
+.. |SiteItem| replace:: :javadoc_base_url:`SiteItem <engine/org/craftercms/engine/model/SiteItem.html>`
+
+--------------
+Studio Support
+--------------
+
+Studio support adds authoring tools to your template.  It's important to understand that these macros *ONLY RENDER IN PREVIEW* and *DO NOT* add additional structure to your markup.  A minimal amount of Javascript/css is injected in to your page to facilitate editing tool integration with your page.  
+
+The previous template markup for Studio support still works however the new markup generates cleaner code and exposes new features.
+
+.. note::
+   If your ``cstudio-support.ftl`` uses ``siteContext.overlayCallback`` to check if Engine is running in preview mode, please update your file to use ``modePreview`` instead as ``siteContext.overlayCallback`` and related classes are being discontinued in Engine.
```

Comment:
@sumerjabri the upgrade manager is removing `cstudio-support.ftl` from sites to prevent issues, is this note still needed?